test/QueryPlannerTest.cpp

```diff
@@ -773,6 +773,72 @@ TEST(QueryPlanner, TransitivePathBindRight) {
       ad_utility::testing::getQec("<x> <p> <o>. <x2> <p> <o2>"));
 }
 
+TEST(QueryPlanner, SpatialJoin) {
+  auto scan = h::IndexScanFromStrings;
+  h::expect(
+      "SELECT ?x ?y WHERE {"
+      "?x <p> ?y."
+      "?a <p> ?b."
+      "?y <max-distance-in-meters:1> ?b }",
+      h::SpatialJoin(1, scan("?x", "<p>", "?y"), scan("?a", "<p>", "?b")));
+
+  AD_EXPECT_THROW_WITH_MESSAGE(
+      h::expect(
+          "SELECT ?x ?y WHERE {"
+          "?x <p> ?y."
+          "?a <p> ?b."
+          "?y <max-distance-in-meters:1> ?b ."
+          "?y <a> ?b}",
+          h::SpatialJoin(1, scan("?x", "<p>", "?y"), scan("?a", "<p>", "?b"))),
```

Comment:
I think passing such an explicit matcher here is misleading.
You can choose something like `::testing::_` which matches anything.

Alternatively you can implement (it is rather easy)
`h::expectThrowWithMessage(query, exceptionMessageMatcher)`.